src/tensor_ops/upscale2d/upscale2d.cu

```diff
@@ -0,0 +1,243 @@
+#include "cuda_utils.cuh"
+
+struct Upscale2dOp {
+    size_t batch;
+    size_t chan;
+    size_t h_in;
+    size_t h_out;
+    size_t w_in;
+    size_t w_out;
+};
+
+template<typename T>
+__device__ void nearest_upscale2d_fwd(
+    const Upscale2dOp op,
+    const size_t *inp_strides,
+    const size_t *out_strides,
+    const T *inp, // 4d (Batch, Channels, Height, Width)
+    T *out // 4d (Batch, Channels, HeightOut, WidthOut)
+) {
+    unsigned int i = blockIdx.x * blockDim.x + threadIdx.x;
+    const size_t numel = op.batch * op.chan * op.h_out * op.w_out;
+    if (i >= numel) {
+        return;
+    }
+
+    float h_scale = ((float)(inp_strides[1]/inp_strides[2]))/(out_strides[1]/out_strides[2]);
+    float w_scale = ((float)(inp_strides[2]/inp_strides[3]))/(out_strides[2]/out_strides[3]);
```

Comment:
Have you tried this with broadcasted tensors? Should this be using the shape instead of strides?

Reply:
It should be yes.

Reply:
This should be fixed.